tests/test_tokenizer.py

```diff
@@ -50,17 +50,23 @@ def test_tokenizer_against_hf(config):
 
     assert ours.vocab_size == theirs.vocab_size
     assert ours.vocab_size == config.vocab_size
-    if theirs.bos_token or ours.use_bos:
-        # TODO: may want to reconsider this test. I've found some cases where there isn't a BOS token in the config, HF automatically sets to None, but the exist flow,
-        # even if it's used, will have it default ours to 1. (see https://huggingface.co/tiiuae/falcon-7b/blob/main/generation_config.json#L3)
+
+    if config.name.startswith("falcon") or config.name.startswith("stablecode"):
```

Comment:
Also nice! FYI I submitted a PR on the falcon config files on huggingface, which should fix the issue upstream as well (though it is not clear when that changes if ever) If it does indeed get fixed there, can adjust it here as well.

Reply:
What's the link? I'd like to see it

Reply:
This, https://huggingface.co/tiiuae/falcon-rw-7b/discussions/4/files, copy and pasted for all falcon variants aside from 180B. 

Looks like the problem was already fixed there: 
https://huggingface.co/tiiuae/falcon-180B-chat/blob/main/generation_config.json

Reply:
You seem to have submitted it to a different model. We use https://huggingface.co/tiiuae/falcon-7b (not `falcon-rw-7b`)